packages/rs-drive-abci/src/execution/platform_events/core_chain_lock/make_sure_core_is_synced_to_chain_lock/v0/mod.rs

```diff
@@ -0,0 +1,20 @@
+use crate::error::Error;
+use crate::platform_types::platform::Platform;
+use crate::rpc::core::CoreRPCLike;
+use dashcore_rpc::dashcore::ChainLock;
+use dpp::version::PlatformVersion;
+
+impl<C> Platform<C>
+where
+    C: CoreRPCLike,
+{
+    /// The point of this call is to make sure core is synced.
+    /// Before this call we had previously validated that the chain lock is valid.
+    pub(super) fn make_sure_core_is_synced_to_chain_lock_v0(
+        &self,
+        chain_lock: &ChainLock,
+    ) -> Result<bool, Error> {
+        // We need to make sure core is synced to the core height we see as valid for the state transitions
+        Ok(self.core_rpc.submit_chain_lock(chain_lock)?)
```

Comment:
Does this command block the execution until core block height is equal or grater than chain lock block height?